### PR TITLE
finished express-notes-api

### PR DESCRIPTION
**Pull request by byu014**
GET
expressnotesapiget

POST
expressnotesapipost

DELETE
expressnotesapidelete

PUT
expressnotesapiput

### CODE REVIEW

File: express-notes-api/index.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  res.status(200).json(notesArr);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  const { notes } = data;
+  const id = req.params.id;
+  if (Number(id) < 0) {
+    res.status(400).json({ error: 'id must be a positive integer' });
+    return;
+  }
```
Comment (TimelikeClosure) [inline express-notes-api/index.js:29]: `id`s with a decimal (ex: `4.252`) should also produce a `400 Bad Request`, as they aren't integers.

File: express-notes-api/index.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+  if (notes[id]) {
+    res.status(200).json(notes[id]);
+  } else {
+    res.status(400).json({ error: `cannot find note with id ${id}` });
+  }
```
Comment (TimelikeClosure) [inline express-notes-api/index.js:35]: Non-existent (but valid) `id`s should produce a `404 Not Found`:

---

image

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]

File: express-notes-api/index.js (hunk 1 above)
Comment (byu014) [reply express-notes-api/index.js]: i updated it to check for decimals as well as fixing that status code

File: express-notes-api/index.js [hunk 3]
```diff
@@ -0,0 +1,109 @@
+const express = require('express');
+const fs = require('fs');
+const app = express();
+let data = null;
+fs.readFile('./data.json', 'utf-8', (err, dataJSON) => {
+  if (err) {
+    process.exit(1);
+  }
```
Comment (TimelikeClosure) [inline express-notes-api/index.js:8]: Not a requirement for the exercise, but if it fails to read the file on start-up, you may want to print out the error message.

**Review by TimelikeClosure — APPROVED** [review]

**Pull request merged**